src/main/java/teammates/ui/controller/InstructorFeedbackEditPageData.java

```diff
@@ -128,19 +139,79 @@ private void buildExistingQuestionForm(String feedbackSessionName,
         for (FeedbackParticipantType participantType : question.showResponsesTo) {
             isResponsesVisibleFor.put(participantType.name(), true);
         }
-        FeedbackQuestionVisibilitySettings visibilitySettings = new FeedbackQuestionVisibilitySettings(
-                                                                        question.getVisibilityMessage(),
-                                                                        isResponsesVisibleFor,
-                                                                        isGiverNameVisibleFor,
-                                                                        isRecipientNameVisibleFor);
-        qnForm.setVisibilitySettings(visibilitySettings);
-        
-        qnForm.setQuestionHasResponses(questionHasResponses.get(question.getId()));
-        
-        qnForm.setQuestionSpecificEditFormHtml(questionDetails.getQuestionSpecificEditFormHtml(questionIndex));
-        qnForm.setEditable(false);
-        
-        qnForms.add(qnForm);
+
+        String dropdownMenuLabel = getDropdownMenuLabel(question);
+
+        return new FeedbackQuestionVisibilitySettings(question.getVisibilityMessage(), isResponsesVisibleFor,
+                                                      isGiverNameVisibleFor, isRecipientNameVisibleFor, dropdownMenuLabel);
+    }
+
+    private String getDropdownMenuLabel(FeedbackQuestionAttributes question) {
+        if (isVisibilitySetToAnonymousToRecipientAndInstructors(question)) {
+            return Const.FeedbackQuestion.COMMON_VISIBILITY_OPTIONS.get("ANONYMOUS_TO_RECIPIENT_AND_INSTRUCTORS");
+        }
+
+        if (isVisibilitySetToAnonymousToRecipientVisibleToInstructors(question)) {
+            return Const.FeedbackQuestion.COMMON_VISIBILITY_OPTIONS.get("ANONYMOUS_TO_RECIPIENT_VISIBLE_TO_INSTRUCTORS");
+        }
+
+        if (isVisibilitySetToVisibleToInstructorsOnly(question)) {
+            return Const.FeedbackQuestion.COMMON_VISIBILITY_OPTIONS.get("VISIBLE_TO_INSTRUCTORS_ONLY");
+        }
+
+        if (isVisibilitySetToVisibleToRecipientAndInstructors(question)) {
+            return Const.FeedbackQuestion.COMMON_VISIBILITY_OPTIONS.get("VISIBLE_TO_RECIPIENT_AND_INSTRUCTORS");
+        }
+
+        return "Custom visibility option:";
```

Comment:
This is singular because it is showing what the user has chosen.
The one in the dropdown menu should be plural though: `Custom visibility options...`